lib/types.js

```diff
@@ -470,21 +469,45 @@ module.exports = function (expect) {
         name: 'array-like',
         base: 'object',
         identify: false,
+        numericalPropertiesOnly: true,
         getKeys: function (obj) {
             var keys = new Array(obj.length);
             for (var i = 0 ; i < obj.length ; i += 1) {
                 keys[i] = i;
             }
+            if (!this.numericalPropertiesOnly) {
+                Object.keys(obj).forEach(function (key) {
+                    if (!numericalRegExp.test(key)) {
+                        keys.push(key);
+                    }
+                });
+            }
             return keys;
         },
         equal: function (a, b, equal) {
             if (a === b) {
                 return true;
             } else if (a.constructor === b.constructor && a.length === b.length) {
-                for (var i = 0; i < a.length; i += 1) {
-                    if (!equal(a[i], b[i])) {
+                var aType = expect.findTypeOf(a);
+                var bType = expect.findTypeOf(b);
```

Comment:
This is a really had one, we should probably just have given you the expect instance as an argument - but it is a little too late for that now. I don't really know what to do about this :-S